src/logging.cc

```diff
@@ -66,82 +74,133 @@ Logger::Flush()
-      stream_ << system_time.wYear << '-' << std::setfill('0') << std::setw(2)
-              << system_time.wMonth << '-' << std::setw(2) << system_time.wDay
-              << 'T' << std::setw(2) << system_time.wHour << ':' << std::setw(2)
-              << system_time.wMinute << ':' << std::setw(2)
-              << system_time.wSecond << "Z "
-              << level_name_[std::min(level, (uint32_t)Level::kINFO)] << ' '
-              << static_cast<uint32_t>(GetCurrentProcessId()) << ' ' << path
-              << ':' << line << "] ";
-#else
-      struct timeval tv;
-      gettimeofday(&tv, NULL);
-      struct tm tm_time;
-      gmtime_r(((time_t*)&(tv.tv_sec)), &tm_time);
-      stream_ << (tm_time.tm_year + 1900) << '-' << std::setfill('0')
-              << std::setw(2) << (tm_time.tm_mon + 1) << '-' << std::setw(2)
-              << tm_time.tm_mday << 'T' << std::setw(2) << tm_time.tm_hour
-              << ':' << std::setw(2) << tm_time.tm_min << ':' << std::setw(2)
-              << tm_time.tm_sec << "Z "
-              << level_name_[std::min(level, (uint32_t)Level::kINFO)] << ' '
-              << static_cast<uint32_t>(getpid()) << ' ' << path << ':' << line
-              << "] ";
+      stream << (tm_time.tm_year + 1900) << '-' << std::setfill('0')
+             << std::setw(2) << (tm_time.tm_mon + 1) << '-' << std::setw(2)
+             << tm_time.tm_mday << 'T' << std::setw(2) << tm_time.tm_hour << ':'
+             << std::setw(2) << tm_time.tm_min << ':' << std::setw(2)
+             << tm_time.tm_sec << "Z";
+      break;
+    }
+    case Logger::Format::kJSONL: {
+      stream << (tm_time.tm_year + 1900) << '-' << std::setfill('0')
+             << std::setw(2) << (tm_time.tm_mon + 1) << '-' << std::setw(2)
+             << tm_time.tm_mday << 'T' << std::setw(2) << tm_time.tm_hour << ':'
+             << std::setw(2) << tm_time.tm_min << ':' << std::setw(2)
+             << tm_time.tm_sec << '.' << std::setw(6) << timestamp_.tv_usec
+             << "Z";
+      break;
+    }
+  }
+}
+
 #endif
+
+void
+LogMessage::LogPreamble(std::stringstream& stream)
+{
+  switch (gLogger_.LogFormat()) {
+    case Logger::Format::kDEFAULT: {
+      stream << LEVEL_NAMES_[level_][0];
+      LogTimestamp(stream);
+      stream << ' ' << pid_ << ' ' << path_ << ':' << line_ << "] ";
+
+      break;
+    }
+    case Logger::Format::kISO8601: {
+      LogTimestamp(stream);
+      stream << " " << LEVEL_NAMES_[level_][0] << ' ' << pid_ << ' ' << path_
+             << ':' << line_ << "] ";
+      break;
+    }
+    case Logger::Format::kJSONL: {
       break;
     }
   }
 }
 
+
 LogMessage::~LogMessage()
 {
-  gLogger_.Log(stream_.str());
-}
+  //  gLogger_.SetLogFormat(Logger::Format::kJSONL);
 
+  switch (gLogger_.LogFormat()) {
+    case Logger::Format::kDEFAULT:
+    case Logger::Format::kISO8601: {
+      std::stringstream preamble;
+      LogPreamble(preamble);
+      //     std::string escaped(message_.str());
```

Comment:
```suggestion
```

Reply:
left for easy debug until we settle on direction - but good to keep track